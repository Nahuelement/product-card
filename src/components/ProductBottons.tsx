import {  useContext} from 'react';
import { ProductContext } from './ProductCard';
import styles from '../styles/styles.module.css'
import React from 'react'
// import { useDispatch, useSelector } from 'react-redux'

export interface PropsButton {
    className?:string,
    style?:React.CSSProperties,
    product?:any,
    quantity?:number
}

export const ProductBottons = ({className,style,product,quantity}:PropsButton) =>{

    
    const {counter, increaseBy,isMaxCountReached } = useContext(ProductContext)
   console.log(product,quantity)

    // const dispatch = useDispatch()
    

    // const isMaxCount = useCallback(() => {
    //    if(counter===maxCount) return true
    //    else return false
    //  }, [counter,maxCount]);
     
    
    
   return(
       <div 
       className = {`${styles.buttonsContainer} ${className}`}
       style = {style}

       >
       <button 
       className ={  styles.buttonMinus}
     
       
       onClick= {() =>{
        // dispatch({type:'put',payload:product})
        increaseBy(-1)}
       } 
       >
           -
       </button>
       <div 
       className ={styles.countLabel}>
           {counter}
       </div>
       
       <button 
       className ={ ` ${ styles.buttonAdd } ${isMaxCountReached && styles.disabled}  ` }
        disabled = {isMaxCountReached}
       onClick= {() => {
        //    dispatch({type:'add',payload:product})
           increaseBy(+1)}}
       >
           +
       </button>

   </div>
   )
}