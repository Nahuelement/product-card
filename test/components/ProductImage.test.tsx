import  renderer  from 'react-test-renderer'
import { ProductImage } from '../../src/components/ProductImage';
import { product2 } from '../data/product';
import { ProductCard } from '../../src/components/ProductCard';
import React from 'react'




describe('ProductImage', () => { 
    
    test('Probando visicualizacion de imagen  ', () => { 
        const wrapper = renderer.create(
            <ProductImage img='https://i.blogs.es/6f44dd/google-2015-1/450_1000.jpg'/>
        )
    
        expect(wrapper.toJSON()).toMatchSnapshot()
    }) 
    test('Probando componente al interiro de productCard pero sin imagen', () => {

        const wrapper = renderer.create(

            <ProductCard product = {product2} >
                {
                    ()=>(
                    <ProductImage />
                    )
                }
            </ProductCard>
        )
        
        expect(wrapper.toJSON()).toMatchSnapshot()
         })
        })