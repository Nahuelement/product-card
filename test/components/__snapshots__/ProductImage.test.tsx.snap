// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Probando componente al interiro de productCard pero sin imagen 1`] = `
<div
  className="productCard undefined "
>
  <img
    alt="Coffee Mug"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage Probando visicualizacion de imagen   1`] = `
<img
  alt="Coffee Mug"
  className="productImg undefined"
  src="https://i.blogs.es/6f44dd/google-2015-1/450_1000.jpg"
/>
`;
